app/containers/AccountProvider/saga.js

```diff
@@ -176,6 +243,17 @@ export function* updateAccWorker({ eos }) {
     }
 
     if (account) {
+      const { user } = profileInfo;
+      const cookieName = `${REFERRAL_REWARD_SENT}_${user}`;
+      if (
+        profileInfo.pay_out_rating > REFERRAL_REWARD_RATING &&
+        !getCookie(cookieName)
+      ) {
+        yield call(rewardRefer, user, eos);
+        setCookie({ name: cookieName, value: true });
```

Comment:
Cookie should be set only if operation `rewardRefer` completed successfully.

Reply:
Fixed